### PR TITLE
making cronjob fence-visa-update optional

### PR DESCRIPTION
**Pull request by ajoaugustine**
Jira Ticket: [GPE-891](https://ctds-planx.atlassian.net/browse/GPE-891)

### Improvements
making cronjob fence-visa-update optional

### Deployment changes
Use ENABLE_FENCE_VISA_UPDATE_CRON in the fence manifest (fence-config-public.yaml) to make cronjob fence-visa-update optional. ENABLE_FENCE_VISA_UPDATE_CRON can be set to true or false.
example:
ENABLE_FENCE_VISA_UPDATE_CRON: true
ENABLE_FENCE_VISA_UPDATE_CRON: false
The current job status remains unaffected if ENABLE_FENCE_VISA_UPDATE_CRON is not set.

[GPE-891]: https://ctds-planx.atlassian.net/browse/GPE-891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: gen3/bin/kube-setup-fence.sh
Comment (paulineribeyre) [inline gen3/bin/kube-setup-fence.sh:100]: i think we should default to either true or false 🤔 i would rather disable it by default to reduce the noise and so it won't be set up when we don't need it, but since it's current automatically enabled, it would be a breaking change. We would need to coordinate (Commons that need the job would need to add `ENABLE_FENCE_VISA_UPDATE_CRON: true` to the fence config before we merge this PR). @Avantol13 or @BinamB might have an opinion?

Comment (BinamB) [reply gen3/bin/kube-setup-fence.sh:100]: I would say we should disable it by default too. I think its been running and failing in places its not supposed to be run in.

Comment (Avantol13) [reply gen3/bin/kube-setup-fence.sh:100]: `GA4GH_PASSPORTS_TO_DRS_ENABLED`

Comment (Avantol13) [reply gen3/bin/kube-setup-fence.sh:100]: hmm maybe this should be separate though, that only disables the ability to POST passposts to the DRS endpoint..

Comment (Avantol13) [reply gen3/bin/kube-setup-fence.sh:100]: ultimately I think the logic of whether we actually sync visas is here: https://github.com/uc-cdis/fence/blame/master/fence/blueprints/login/ras.py#L47-L59

Comment (Avantol13) [reply gen3/bin/kube-setup-fence.sh:100]: so if we end up adding a new config, we need to make sure that we error with weird combinations like: the above forcing global visa sync but the new config is false, so the cron would never run. That would be really really bad

Comment (paulineribeyre) [reply gen3/bin/kube-setup-fence.sh]: @Avantol13 Here is a Fence PR to add the `ENABLE_VISA_UPDATE_CRON` setting + validation around visa parsing: https://github.com/uc-cdis/fence/pull/1087

**Review by paulineribeyre — COMMENTED** [review]
lgtm

but we can't merge until the Fence change is merged and the systems that need visa parsing add the `ENABLE_VISA_UPDATE_CRON` config AND deploy the new version of fence

**Review by paulineribeyre — APPROVED** [review]
@k-burt-uch confirmed that BDC and DCF do not need config updates

**Pull request merged**
